core/src/main/java/org/apache/gravitino/utils/ClientPoolImpl.java

```diff
@@ -131,10 +131,11 @@ public void close() {
   }
 
   private C get() throws InterruptedException {
-    Preconditions.checkState(!closed, "Cannot get a client from a closed pool");
     while (true) {
+      Preconditions.checkState(!closed, "Cannot get a client from a closed pool");
       if (!clients.isEmpty() || currentSize < poolSize) {
         synchronized (this) {
+          Preconditions.checkState(!closed, "Cannot get a client from a closed pool");
```

Comment:
While this works, I'm not sure that using a precondition is the best approach.

Reply:
The value `closed` will be changed in the method `close`, which is not protected by the object lock, so what the   benefit of checking it again and again in the synchronized code blocks?